### PR TITLE
feat: new leverage with opensea sell support

### PR DESCRIPTION
**Pull request by uzielflorida**

### CODE REVIEW

**Review by franco-gondi — APPROVED** [review]

File: src/gondi.ts [hunk 1]
```diff
@@ -794,7 +794,7 @@ export class Gondi {
         throw new Error(`No native bid for price ${price}`);
       }
 
-      executionData = await this.reservoir.generateMatchOrdersExecutionData({
```
Comment (mishuagopian) [inline src/gondi.ts:797]: Delete `generateMatchOrdersExecutionData` then

File: src/reservoir/Reservoir.ts [hunk 2]
```diff
@@ -211,6 +212,67 @@ export class Reservoir {
...
+    const nftWithCriteriaIndex = order.parameters.consideration.findIndex(
+      (consid) => consid.itemType === 4
+    );
+
+    const fulfillOrderCallbackData = encodeFunctionData({
+      abi: seaportABI,
+      functionName: "fulfillAdvancedOrder",
+      args: [
+        order,
+        nftWithCriteriaIndex !== -1
```
Comment (mishuagopian) [inline src/reservoir/Reservoir.ts:246]: Should we fail in this case?

File: src/gondi.ts (hunk 1 above)
Comment (uzielflorida) [reply src/gondi.ts:797]: match orders is still used for bnpl

File: src/reservoir/Reservoir.ts (hunk 2 above)
Comment (uzielflorida) [reply src/reservoir/Reservoir.ts:246]: There are some bids that do not have an NftWithCriteria, like private bids

**Review by mishuagopian — APPROVED** [review]

**Pull request merged**